jib-core/src/main/java/com/google/cloud/tools/jib/ncache/LastModifiedTimeMetadata.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.ncache;
+
+import com.google.cloud.tools.jib.blob.Blob;
+import com.google.cloud.tools.jib.blob.Blobs;
+import com.google.cloud.tools.jib.image.LayerEntry;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.attribute.FileTime;
+import java.time.Instant;
+import java.util.Optional;
+
+/**
+ * Metadata that is the latest last modified time of all the source files in a list of {@link
+ * LayerEntry}s.
```

Comment:
Since we now have `getLastModifiedTime(CacheEntry)`, this Javadoc (saying "time of all the source files ... of `LayerEntry`s") doesn't fit very well.

Reply:
This javadoc is referring to the actual contents of the metadata, which is the last modified time of layer entries. Getting the last modified time from a `CacheEntry` is just deserializing the metadata, which still is the last modified time of the layer entries that made up the layer.

Reply:
But all the methods are just static util methods. There is no field, and the class is stateless, not representing anything actually. So I'm not sold that this class is referring to the actual contents of the metadata, since there is no metadata actually associated with this class.

Reply:
Hmm, I'll try rewording this. This is intended as a serialization/deserialization helper for metadata that stores the last modified time, in comparison to another possible `...Metadata` class that could serialize/deserialize for different contents.